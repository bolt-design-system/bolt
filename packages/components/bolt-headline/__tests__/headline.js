import { render, stopServer, html } from '../../../testing/testing-helpers';

const { readYamlFileSync } = require('@bolt/build-tools/utils/yaml');
const { join } = require('path');
const schema = readYamlFileSync(join(__dirname, '../headline.schema.yml'));
const { tag, size, align, transform } = schema.properties;

describe('<bolt-headline> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  test('basic usage headline', async () => {
    const results = await render('@bolt-components-headline/headline.twig', {
      text: 'this is a headline',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage subheadline', async () => {
    const results = await render('@bolt-components-headline/subheadline.twig', {
      text: 'this is an subheadline',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage eyebrow', async () => {
    const results = await render('@bolt-components-headline/eyebrow.twig', {
      text: 'this is an eyebrow',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
  test('basic usage text', async () => {
    const results = await render('@bolt-components-headline/text.twig', {
      text: 'this is text',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
  test('basic usage lead', async () => {
    const results = await render('@bolt-components-headline/lead.twig', {
      text: 'this is a lead',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage headline with attributes', async () => {
    const results = await render('@bolt-components-headline/headline.twig', {
      attributes: {
        'foo bar': 'baz',
      },
      text: 'this is a headline',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  tag.enum.forEach((displayChoice) => {
    test(`tag display: ${displayChoice}`, async () => {
      const results = await render('@bolt-components-headline/headline.twig', {
        tag: displayChoice,
        text: 'Some text',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  align.enum.forEach((alignmentChoice) => {
    test(`text alignment: ${alignmentChoice}`, async () => {
      const results = await render('@bolt-components-headline/headline.twig', {
        align: alignmentChoice,
        text: 'Some text',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  size.enum.forEach((sizeChoice) => {
    const fontWeight = ['bold', 'regular', 'semibold'];

    fontWeight.forEach((weightChoice) => {
      test(`Sizes at all varient font weights: ${sizeChoice}, ${weightChoice}`, async () => {
        const results = await render(
          '@bolt-components-headline/headline.twig',
          {
            size: sizeChoice,
            weight: weightChoice,
            text: 'Some text',
          },
        );
        expect(results.ok).toBe(true);
        expect(results.html).toMatchSnapshot();
      });
    });
  });

  size.enum.forEach((sizeChoice) => {
    const fontStyle = ['normal', 'italic'];

    fontStyle.forEach((fontChoice) => {
      test(`Sizes at all varient font styles: ${sizeChoice}, ${fontChoice}`, async () => {
        const results = await render(
          '@bolt-components-headline/headline.twig',
          {
            size: sizeChoice,
            style: fontChoice,
            text: 'Some text',
          },
        );
        expect(results.ok).toBe(true);
        expect(results.html).toMatchSnapshot();
      });
    });
  });

  transform.enum.forEach((caseChoice) => {
    test(`text casing: ${caseChoice}`, async () => {
      const results = await render('@bolt-components-headline/headline.twig', {
        transform: caseChoice,
        text: 'Some text',
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test('headline with associated link', async () => {
    const results = await render('@bolt-components-headline/headline.twig', {
      url: 'www.testurl.com',
      text: 'this is a headline',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('headline with icon', async () => {
    const results = await render('@bolt-components-headline/headline.twig', {
      icon: 'check',
      text: 'this is a headline',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
