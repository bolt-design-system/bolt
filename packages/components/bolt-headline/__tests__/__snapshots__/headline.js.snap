// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-headline> Component <span>-based Headlines do not overlap with neighboring <p> tags 1`] = `
<span class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  This is a Headline
</span>
`;

exports[`<bolt-headline> Component <span>-based Headlines do not overlap with neighboring <p> tags 2`] = `
<span
  class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge"
  style=""
>
  This is a Headline
</span>
`;

exports[`<bolt-headline> Component <span>-based Headlines horizontally align properly. 1`] = `
<span class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--center c-bolt-headline--xlarge">
  This is a Headline
</span>
`;

exports[`<bolt-headline> Component <span>-based Headlines horizontally align properly. 2`] = `
<span
  class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--center c-bolt-headline--xlarge"
  style=""
>
  This is a Headline
</span>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: large, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--large">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: large, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--large">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: medium, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--medium">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: medium, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--medium">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: small, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--small c-bolt-headline--uppercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: small, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--small c-bolt-headline--uppercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xlarge, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xlarge, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xsmall, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xsmall, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxlarge, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxlarge, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxsmall, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxsmall, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxxlarge, italic 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font styles: xxxlarge, normal 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: large, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--large">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: large, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--large">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: large, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--large">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: medium, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--medium">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: medium, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--medium">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: medium, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--medium">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: small, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--small c-bolt-headline--uppercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: small, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--small c-bolt-headline--uppercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: small, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--small c-bolt-headline--uppercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xlarge, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xlarge, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xlarge, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xsmall, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xsmall, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--xsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xsmall, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--xsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxlarge, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxlarge, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--xxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxlarge, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--xxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxsmall, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxsmall, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--xxsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxsmall, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--xxsmall">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxxlarge, bold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xxxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxxlarge, regular 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--regular c-bolt-headline--xxxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component Sizes at all varient font weights: xxxlarge, semibold 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--semibold c-bolt-headline--xxxlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component basic usage eyebrow 1`] = `
<p class="c-bolt-eyebrow c-bolt-eyebrow--regular">
  this is an eyebrow
</p>
`;

exports[`<bolt-headline> Component basic usage headline 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  this is a headline
</h2>
`;

exports[`<bolt-headline> Component basic usage headline with attributes 1`] = `
<h2 foo
    bar="baz"
    class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge"
>
  this is a headline
</h2>
`;

exports[`<bolt-headline> Component basic usage lead 1`] = `
<p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--large">
  this is a lead
</p>
`;

exports[`<bolt-headline> Component basic usage subheadline 1`] = `
<p class="c-bolt-subheadline c-bolt-subheadline--regular c-bolt-subheadline--xlarge">
  this is an subheadline
</p>
`;

exports[`<bolt-headline> Component basic usage text 1`] = `
<p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--medium">
  this is text
</p>
`;

exports[`<bolt-headline> Component headline with associated link 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge c-bolt-headline--link">
  <a href="www.testurl.com"
     class="e-bolt-text-link e-bolt-text-link--reversed-underline"
  >
    this is a headline
    <span class="e-bolt-text-link__icon-after"
          aria-hidden="true"
    >
      <bolt-icon name="chevron-right"
                 aria-hidden="true"
      >
        <span class="c-bolt-icon c-bolt-icon--chevron-right">
          <svg class="c-bolt-icon__icon">
            <use xlink:href="#chevron-right">
            </use>
          </svg>
        </span>
      </bolt-icon>
    </span>
  </a>
</h2>
`;

exports[`<bolt-headline> Component headline with icon 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  <span class="c-bolt-headline__text">
    this is a headline
    <span class="c-bolt-headline__icon c-bolt-headline__icon--position-after">
      <bolt-icon name="check"
                 aria-hidden="true"
      >
        <span class="c-bolt-icon c-bolt-icon--check">
          <svg class="c-bolt-icon__icon">
            <use xlink:href="#check">
            </use>
          </svg>
        </span>
      </bolt-icon>
    </span>
  </span>
</h2>
`;

exports[`<bolt-headline> Component tag display: cite 1`] = `
<cite class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</cite>
`;

exports[`<bolt-headline> Component tag display: div 1`] = `
<div class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</div>
`;

exports[`<bolt-headline> Component tag display: h1 1`] = `
<h1 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h1>
`;

exports[`<bolt-headline> Component tag display: h2 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component tag display: h3 1`] = `
<h3 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h3>
`;

exports[`<bolt-headline> Component tag display: h4 1`] = `
<h4 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h4>
`;

exports[`<bolt-headline> Component tag display: h5 1`] = `
<h5 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h5>
`;

exports[`<bolt-headline> Component tag display: h6 1`] = `
<h6 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</h6>
`;

exports[`<bolt-headline> Component tag display: p 1`] = `
<p class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</p>
`;

exports[`<bolt-headline> Component tag display: span 1`] = `
<span class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge">
  Some text
</span>
`;

exports[`<bolt-headline> Component text alignment: center 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--center c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component text alignment: left 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--left c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component text alignment: right 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--right c-bolt-headline--xlarge">
  Some text
</h2>
`;

exports[`<bolt-headline> Component text casing: capitalize 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge c-bolt-headline--capitalize">
  Some text
</h2>
`;

exports[`<bolt-headline> Component text casing: lowercase 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge c-bolt-headline--lowercase">
  Some text
</h2>
`;

exports[`<bolt-headline> Component text casing: uppercase 1`] = `
<h2 class="c-bolt-headline c-bolt-headline--bold c-bolt-headline--xlarge c-bolt-headline--uppercase">
  Some text
</h2>
`;
